pkg/tenant/db/tenants.go

```diff
@@ -30,18 +31,16 @@ func (t *Tenant) Namespace() string {
 }
 
 func (t *Tenant) MarshalValue() ([]byte, error) {
-	// TODO: look into bson, msgpack, etc.
-	return json.Marshal(t)
+	return msgpack.Marshal(t)
 }
 
 func (t *Tenant) UnmarshalValue(data []byte) error {
-	// TODO: look into bson, msgpack, etc.
-	return json.Unmarshal(data, t)
+	return msgpack.Unmarshal(data, t)
 }
 
 func CreateTenant(ctx context.Context, tenant *Tenant) (err error) {
-	if tenant.ID == uuid.Nil {
-		tenant.ID = uuid.New()
+	if tenant.ID.String() == "" {
```

Comment:
ulid.Nil isn't provided by the ulid package, so I chose to go this route. However, I could change the code and follow the example here: https://github.com/oklog/ulid/issues/84#issuecomment-1254728179